src/app/commands/cmd_color_quantization.cpp

```diff
@@ -31,11 +33,70 @@
 
 namespace app {
 
+#if ENABLE_UI
+class PaletteFromSpriteWindow : public app::gen::PaletteFromSprite {
+public:
+  PaletteFromSpriteWindow(Site& site,
+                          PalettePicks& entries,
+                          Preferences& pref)
+    : m_specificNumber(true)
+    , m_colorLimit(256)
+    , m_replacePalette(false)
+    , m_replaceCurrentRange(false)
+    , m_withAlpha(pref.quantization.withAlpha())
+    , m_mapAlgorithm(pref.quantization.mappingAlgorithm())
+
+  {
+    colorMappingAlgorithm()->Change.connect(base::Bind<void>(&PaletteFromSpriteWindow::onChangeMapAlgorithm, this));
+    colorMappingAlgorithm()->setSelectedItemIndex((int)m_mapAlgorithm);
+
+    if (site.document()->sprite()->pixelFormat() == PixelFormat::IMAGE_RGB) {
+      if (m_mapAlgorithm == MapAlgorithm::OCTREE) {
+        alphaChannel()->setSelected(false);
+        alphaChannel()->setEnabled(false);
+      }
+      else {
+        alphaChannel()->setEnabled(true);
+        alphaChannel()->setSelected(m_withAlpha);
+      }
+    }
+    else {
+      colorMappingAlgorithm()->setSelectedItemIndex((int)MapAlgorithm::RGBA);
+      colorMappingAlgorithm()->setEnabled(false);
+    }
+
+  }
+
+  ~PaletteFromSpriteWindow() {}
+
+  void onChangeMapAlgorithm() {
+    MapAlgorithm selectedAlgo = (MapAlgorithm)colorMappingAlgorithm()->getSelectedItemIndex();
+    if (selectedAlgo == MapAlgorithm::OCTREE) {
+      m_mapAlgorithm = MapAlgorithm::OCTREE;
+      alphaChannel()->setSelected(false);
+      alphaChannel()->setEnabled(false);
+    }
+    else {
+      m_mapAlgorithm = MapAlgorithm::RGBA;
+      alphaChannel()->setEnabled(true);
+    }
+  }
+
+  bool m_specificNumber;
+  int m_colorLimit = 256;
+  bool m_replacePalette;
+  bool m_replaceCurrentRange;
+  bool m_withAlpha;
+  MapAlgorithm m_mapAlgorithm;
+};
+#endif
+
 struct ColorQuantizationParams : public NewParams {
   Param<bool> ui { this, true, "ui" };
   Param<bool> withAlpha { this, true, "withAlpha" };
   Param<int> maxColors { this, 256, "maxColors" };
   Param<bool> useRange { this, false, "useRange" };
+  Param<MapAlgorithm> mappingAlgorithm { this, MapAlgorithm::RGBA, "mappingAlgorithm" };
```

Comment:
The name of this parameter must match the name of the `ChangePixelFormatCommand` parameter.